tensorrt_llm/_torch/pyexecutor/model_engine.py

```
@@ -559,7 +559,144 @@ def warmup(self, resource_manager: ResourceManager) -> None:
+        def get_cuda_graph_warmup_request(batch_size):
+            available_blocks = kv_cache_manager.get_num_free_blocks()
+            if available_blocks >= batch_size:
+                result = ScheduledRequests()
+                result.context_requests = []
+                # Add (batch_size - 1) dummy requests with seq_len=1.
+                # Should only need one more page per request.
+                requests = kv_cache_manager.add_dummy_requests(
+                    list(range(batch_size - 1)),
+                    is_gen=True,
+                    max_num_draft_tokens=self.max_draft_len,
+                    use_mrope=use_mrope,
```

Ruff (0.14.5)

Undefined name `use_mrope`

(F821)